homes-ms-storesmanagement/src/test/java/com/tenx/ms/storesmanagement/products/rest/ProductControllerTest.java

```diff
@@ -0,0 +1,189 @@
+package com.tenx.ms.storesmanagement.products.rest;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.tenx.ms.commons.config.Profiles;
+import com.tenx.ms.commons.rest.RestConstants;
+import com.tenx.ms.commons.rest.dto.ResourceCreated;
+import com.tenx.ms.commons.tests.AbstractIntegrationTest;
+import com.tenx.ms.storesmanagement.StoresmanagementServiceApp;
+import com.tenx.ms.storesmanagement.products.rest.dto.ProductDTO;
+import org.apache.commons.io.FileUtils;
+import org.flywaydb.test.annotation.FlywayTest;
+import org.flywaydb.test.junit.FlywayTestExecutionListener;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.TestRestTemplate;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.support.DependencyInjectionTestExecutionListener;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.greaterThan;
+import static org.springframework.test.util.AssertionErrors.assertEquals;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = StoresmanagementServiceApp.class)
+@ActiveProfiles(Profiles.TEST_NOAUTH)
+@TestExecutionListeners({DependencyInjectionTestExecutionListener.class, FlywayTestExecutionListener.class})
+public class ProductControllerTest extends AbstractIntegrationTest {
+
+
+    private final RestTemplate template = new TestRestTemplate();
+
+    private static final Long NOT_VALID_ID = 11111L;
+    private static final Long VALID_ID = 1L;
+
+    private static final Long NOT_VALID_STORE_ID = 11111L;
+    private static final Long VALID_STORE_ID = 1L;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    private final ClassLoader CLASSLOADER = Thread.currentThread().getContextClassLoader();
+
+    @Value("testData/success/product/successTestProduct.json")
```

Comment:
You can specify @Value("classpath:testData/success/product/successTestProduct.json"), then you don't need to use the ClassLoader from thread to get the file